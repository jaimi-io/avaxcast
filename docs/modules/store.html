<!DOCTYPE html>
<html class="default no-js">
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title>store | avaxcast-frontend</title>
    <meta name="description" content="Documentation for avaxcast-frontend" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="../assets/css/main.css" />
    <script async src="../assets/js/search.js" id="search-script"></script>
  </head>
  <body>
    <header>
      <div class="tsd-page-toolbar">
        <div class="container">
          <div class="table-wrap">
            <div
              class="table-cell"
              id="tsd-search"
              data-index="../assets/js/search.json"
              data-base=".."
            >
              <div class="field">
                <label
                  for="tsd-search-field"
                  class="tsd-widget search no-caption"
                  >Search</label
                >
                <input id="tsd-search-field" type="text" />
              </div>
              <ul class="results">
                <li class="state loading">Preparing search index...</li>
                <li class="state failure">The search index is not available</li>
              </ul>
              <a href="../index.html" class="title">avaxcast-frontend</a>
            </div>
            <div class="table-cell" id="tsd-widgets">
              <div id="tsd-filter">
                <a
                  href="#"
                  class="tsd-widget options no-caption"
                  data-toggle="options"
                  >Options</a
                >
                <div class="tsd-filter-group">
                  <div class="tsd-select" id="tsd-filter-visibility">
                    <span class="tsd-select-label">All</span>
                    <ul class="tsd-select-list">
                      <li data-value="public">Public</li>
                      <li data-value="protected">Public/Protected</li>
                      <li data-value="private" class="selected">All</li>
                    </ul>
                  </div>
                  <input type="checkbox" id="tsd-filter-inherited" checked />
                  <label class="tsd-widget" for="tsd-filter-inherited"
                    >Inherited</label
                  >
                  <input type="checkbox" id="tsd-filter-externals" checked />
                  <label class="tsd-widget" for="tsd-filter-externals"
                    >Externals</label
                  >
                </div>
              </div>
              <a href="#" class="tsd-widget menu no-caption" data-toggle="menu"
                >Menu</a
              >
            </div>
          </div>
        </div>
      </div>
      <div class="tsd-page-title">
        <div class="container">
          <ul class="tsd-breadcrumb">
            <li>
              <a href="../modules.html">avaxcast-frontend</a>
            </li>
            <li>
              <a href="store.html">store</a>
            </li>
          </ul>
          <h1>Module store</h1>
        </div>
      </div>
    </header>
    <div class="container container-main">
      <div class="row">
        <div class="col-8 col-content">
          <section class="tsd-panel-group tsd-index-group">
            <h2>Index</h2>
            <section class="tsd-panel tsd-index-panel">
              <div class="tsd-index-content">
                <section class="tsd-index-section">
                  <h3>Type aliases</h3>
                  <ul class="tsd-index-list">
                    <li class="tsd-kind-type-alias tsd-parent-kind-module">
                      <a href="store.html#AppDispatch" class="tsd-kind-icon"
                        >App<wbr />Dispatch</a
                      >
                    </li>
                    <li class="tsd-kind-type-alias tsd-parent-kind-module">
                      <a href="store.html#RootState" class="tsd-kind-icon"
                        >Root<wbr />State</a
                      >
                    </li>
                  </ul>
                </section>
                <section class="tsd-index-section">
                  <h3>Variables</h3>
                  <ul class="tsd-index-list">
                    <li class="tsd-kind-variable tsd-parent-kind-module">
                      <a href="store.html#default" class="tsd-kind-icon"
                        >default</a
                      >
                    </li>
                    <li class="tsd-kind-variable tsd-parent-kind-module">
                      <a href="store.html#persistor" class="tsd-kind-icon"
                        >persistor</a
                      >
                    </li>
                  </ul>
                </section>
              </div>
            </section>
          </section>
          <section class="tsd-panel-group tsd-member-group">
            <h2>Type aliases</h2>
            <section
              class="
                tsd-panel tsd-member tsd-kind-type-alias tsd-parent-kind-module
              "
            >
              <a name="AppDispatch" class="tsd-anchor"></a>
              <h3>App<wbr />Dispatch</h3>
              <div class="tsd-signature tsd-kind-icon">
                App<wbr />Dispatch<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-symbol">typeof </span
                ><span class="tsd-signature-type">store.dispatch</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L10"
                      >src/store.ts:10</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
            <section
              class="
                tsd-panel tsd-member tsd-kind-type-alias tsd-parent-kind-module
              "
            >
              <a name="RootState" class="tsd-anchor"></a>
              <h3>Root<wbr />State</h3>
              <div class="tsd-signature tsd-kind-icon">
                Root<wbr />State<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">ReturnType</span
                ><span class="tsd-signature-symbol">&lt;</span
                ><span class="tsd-signature-symbol">typeof </span
                ><span class="tsd-signature-type">store.getState</span
                ><span class="tsd-signature-symbol">&gt;</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L9"
                      >src/store.ts:9</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
          </section>
          <section class="tsd-panel-group tsd-member-group">
            <h2>Variables</h2>
            <section
              class="
                tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module
              "
            >
              <a name="default" class="tsd-anchor"></a>
              <h3><span class="tsd-flag ts-flagConst">Const</span> default</h3>
              <div class="tsd-signature tsd-kind-icon">
                default<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">Store</span
                ><span class="tsd-signature-symbol">&lt;</span
                ><span class="tsd-signature-type">EmptyObject</span
                ><span class="tsd-signature-symbol"> &amp; </span
                ><span class="tsd-signature-symbol">{ </span>isDark<span
                  class="tsd-signature-symbol"
                  >: </span
                ><span class="tsd-signature-type">boolean</span
                ><span class="tsd-signature-symbol">; </span>marketFilter<span
                  class="tsd-signature-symbol"
                  >: </span
                ><a
                  href="../enums/common_enums.Market.html"
                  class="tsd-signature-type"
                  data-tsd-kind="Enumeration"
                  >Market</a
                ><span class="tsd-signature-symbol"> }</span
                ><span class="tsd-signature-symbol"> &amp; </span
                ><span class="tsd-signature-type">PersistPartial</span
                ><span class="tsd-signature-symbol">, </span
                ><a
                  href="../interfaces/lib_global.Action.html"
                  class="tsd-signature-type"
                  data-tsd-kind="Interface"
                  >Action</a
                ><span class="tsd-signature-symbol">&gt;</span
                ><span class="tsd-signature-symbol"> = ...</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L5"
                      >src/store.ts:5</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
            <section
              class="
                tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module
              "
            >
              <a name="persistor" class="tsd-anchor"></a>
              <h3>
                <span class="tsd-flag ts-flagConst">Const</span> persistor
              </h3>
              <div class="tsd-signature tsd-kind-icon">
                persistor<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">Persistor</span
                ><span class="tsd-signature-symbol"> = ...</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L6"
                      >src/store.ts:6</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
          </section>
        </div>
        <div class="col-4 col-menu menu-sticky-wrap menu-highlight">
          <nav class="tsd-navigation primary">
            <ul>
              <li class="">
                <a href="../modules.html">Modules</a>
              </li>
              <li class="current tsd-kind-module">
                <a href="store.html">store</a>
              </li>
            </ul>
          </nav>
          <nav class="tsd-navigation secondary menu-sticky">
            <ul class="before-current">
              <li class="tsd-kind-type-alias tsd-parent-kind-module">
                <a href="store.html#AppDispatch" class="tsd-kind-icon"
                  >App<wbr />Dispatch</a
                >
              </li>
              <li class="tsd-kind-type-alias tsd-parent-kind-module">
                <a href="store.html#RootState" class="tsd-kind-icon"
                  >Root<wbr />State</a
                >
              </li>
              <li class="tsd-kind-variable tsd-parent-kind-module">
                <a href="store.html#default" class="tsd-kind-icon">default</a>
              </li>
              <li class="tsd-kind-variable tsd-parent-kind-module">
                <a href="store.html#persistor" class="tsd-kind-icon"
                  >persistor</a
                >
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <footer class="with-border-bottom">
      <div class="container">
        <h2>Legend</h2>
        <div class="tsd-legend-group">
          <ul class="tsd-legend">
            <li class="tsd-kind-variable">
              <span class="tsd-kind-icon">Variable</span>
            </li>
            <li class="tsd-kind-function">
              <span class="tsd-kind-icon">Function</span>
            </li>
            <li class="tsd-kind-function tsd-has-type-parameter">
              <span class="tsd-kind-icon">Function with type parameter</span>
            </li>
            <li class="tsd-kind-type-alias">
              <span class="tsd-kind-icon">Type alias</span>
            </li>
          </ul>
          <ul class="tsd-legend">
            <li class="tsd-kind-enum">
              <span class="tsd-kind-icon">Enumeration</span>
            </li>
          </ul>
          <ul class="tsd-legend">
            <li class="tsd-kind-interface">
              <span class="tsd-kind-icon">Interface</span>
            </li>
          </ul>
        </div>
      </div>
    </footer>
    <div class="container tsd-generator">
      <p>
        Generated using
        <a href="https://typedoc.org/" target="_blank">TypeDoc</a>
      </p>
    </div>
    <div class="overlay"></div>
    <script src="../assets/js/main.js"></script>
  </body>
</html>
