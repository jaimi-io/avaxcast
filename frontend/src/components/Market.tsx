import Button from "@material-ui/core/Button";
import CheckIcon from "@material-ui/icons/Check";
import CloseIcon from "@material-ui/icons/Close";
import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import {
  createStyles,
  createTheme,
  makeStyles,
  Theme,
  ThemeProvider,
} from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import { useWeb3React } from "@web3-react/core";
import { green, red } from "@material-ui/core/colors";
import { useState, useEffect } from "react";
import {
  buy,
  ContractI,
  getContractInfo,
  getCurrentVotes,
  VotesPerPerson,
} from "common/contract";
import { marketNames } from "common/markets";
import Input from "@material-ui/core/Input";
import { Vote } from "common/enums";
import { fromWei, toBN } from "web3-utils";
import BN from "bn.js";
import { UNDEFINED_NUM_SHARES, UNDEFINED_PRICE } from "common/constants";
import SuccessSnackbar from "./SuccessSnackbar";
import Loading from "./Loading";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    button: {
      margin: theme.spacing(1),
      minHeight: 50,
      minWidth: 150,
    },
  })
);

const yesTheme = createTheme({
  palette: {
    primary: green,
  },
});

const noTheme = createTheme({
  palette: {
    primary: red,
  },
});

interface PropsT {
  address: string;
}

function Market({ address }: PropsT): JSX.Element {
  const classes = useStyles();
  const web3 = useWeb3React();
  const { active } = web3;
  const [loading, setLoading] = useState(false);
  const [openSnackbar, setOpenSnackbar] = useState(false);
  const [success, setSuccess] = useState(false);
  const [isYesVote, setIsYesVote] = useState(true);
  const [contract, setContract] = useState<ContractI>({
    market: 0,
    predictedPrice: "$X",
    date: new Date(),
    volume: 0,
    yesPrice: toBN(UNDEFINED_PRICE),
    noPrice: toBN(UNDEFINED_PRICE),
    address: address,
  });
  const formattedDate = contract.date.toDateString();
  const [numShares, setNumShares] = useState(UNDEFINED_NUM_SHARES);
  const [canBuy, setCanBuy] = useState(false);
  const [totalPrice, setTotalPrice] = useState(toBN(UNDEFINED_PRICE));
  const [currentVotes, setCurrentVotes] = useState<VotesPerPerson>({
    yesVotes: 0,
    noVotes: 0,
  });

  const handleClose = (event?: React.SyntheticEvent, reason?: string) => {
    if (reason === "clickaway") {
      return;
    }
    setOpenSnackbar(false);
  };

  const priceOfShares = (): BN => {
    const pricePerShare = isYesVote ? contract.yesPrice : contract.noPrice;
    return pricePerShare.mul(toBN(numShares));
  };

  const fetchCurrentVotes = async () => {
    if (!active) {
      return;
    }
    const votes = await getCurrentVotes(address, web3);
    setCurrentVotes(votes);
  };

  useEffect(() => {
    getContractInfo(address, setContract);
    fetchCurrentVotes();
  }, []);

  useEffect(() => {
    fetchCurrentVotes();
  }, [active]);

  useEffect(() => {
    setTotalPrice(priceOfShares());
  }, [numShares, isYesVote]);

  useEffect(() => {
    setCanBuy(active && numShares > UNDEFINED_NUM_SHARES && totalPrice.gtn(0));
  }, [active, numShares, totalPrice]);

  return (
    <>
      <Card variant="outlined">
        <CardContent>
          <Grid container justifyContent="center">
            <Grid item xs={6}>
              <Typography component="h1" variant="h6">
                {`Will ${marketNames[contract.market]} reach ${
                  contract.predictedPrice
                } by ${formattedDate}?`}
              </Typography>
            </Grid>
            <Grid item xs={2}>
              <Grid container>
                <Grid item xs={12}>
                  <Typography variant="body2" component="p">
                    {"Predicted Price"}
                  </Typography>
                </Grid>
                <Grid item xs={12}>
                  <Typography component="p">
                    {contract.predictedPrice}
                  </Typography>
                </Grid>
              </Grid>
            </Grid>
            <Grid item xs={2}>
              <Grid container>
                <Grid item xs={12}>
                  <Typography variant="body2" component="p">
                    {"End Date"}
                  </Typography>
                </Grid>
                <Grid item xs={12}>
                  <Typography component="p">{formattedDate}</Typography>
                </Grid>
              </Grid>
            </Grid>
            <Grid item xs={2}>
              <Grid container>
                <Grid item xs={12}>
                  <Typography variant="body2" component="p">
                    {"Total Volume"}
                  </Typography>
                </Grid>
                <Grid item xs={12}>
                  <Typography component="p">{contract.volume}</Typography>
                </Grid>
              </Grid>
            </Grid>
          </Grid>
        </CardContent>
      </Card>
      <Card>
        <CardContent>
          <Grid container justifyContent="center">
            <Grid item xs={6}>
              <ThemeProvider theme={yesTheme}>
                <Grid container>
                  <Grid item xs={3}>
                    <Typography variant="body2" component="p">
                      {"Price"}
                    </Typography>
                    <Typography component="p">{`${fromWei(
                      contract.yesPrice,
                      "ether"
                    )} AVAX`}</Typography>
                  </Grid>
                  <Grid item xs={3}>
                    <Typography variant="body2" component="p">
                      {"Your Votes"}
                    </Typography>
                    <Typography component="p">
                      {currentVotes.yesVotes}
                    </Typography>
                  </Grid>
                </Grid>
                <Button
                  variant="contained"
                  color={isYesVote ? "primary" : "default"}
                  className={classes.button}
                  disabled={!active}
                  onClick={() => setIsYesVote(true)}
                  startIcon={<CheckIcon />}>
                  Yes
                </Button>
              </ThemeProvider>
            </Grid>
            <Grid item xs={6}>
              <ThemeProvider theme={noTheme}>
                <Grid container>
                  <Grid item xs={3}>
                    <Typography variant="body2" component="p">
                      {"Price"}
                    </Typography>
                    <Typography component="p">{`${fromWei(
                      contract.noPrice,
                      "ether"
                    )} AVAX`}</Typography>
                  </Grid>
                  <Grid item xs={3}>
                    <Typography variant="body2" component="p">
                      {"Your Votes"}
                    </Typography>
                    <Typography component="p">
                      {currentVotes.noVotes}
                    </Typography>
                  </Grid>
                </Grid>
                <Button
                  variant="contained"
                  color={isYesVote ? "default" : "primary"}
                  className={classes.button}
                  disabled={!active}
                  onClick={() => setIsYesVote(false)}
                  startIcon={<CloseIcon />}>
                  No
                </Button>
              </ThemeProvider>
            </Grid>
            <Grid item xs={12}>
              <Typography component="p">{"How many shares?"}</Typography>
            </Grid>
            <Grid item xs={12}>
              <Input
                type="number"
                value={numShares}
                onChange={(e) => {
                  const newValue = parseInt(e.target.value);
                  if (
                    (!newValue && newValue !== UNDEFINED_NUM_SHARES) ||
                    newValue < UNDEFINED_NUM_SHARES
                  ) {
                    return;
                  }
                  setNumShares(newValue);
                }}
              />
            </Grid>
            <Grid item xs={12}>
              <Typography component="p">{`Total Price: ${fromWei(
                totalPrice,
                "ether"
              )} AVAX`}</Typography>
            </Grid>
            <Grid item xs={12}>
              <Button
                variant="contained"
                color="primary"
                className={classes.button}
                disabled={!canBuy}
                onClick={async () => {
                  setLoading(true);
                  await buy(
                    contract.address,
                    web3,
                    isYesVote ? Vote.Yes : Vote.No,
                    priceOfShares()
                  )
                    .then(() => {
                      setSuccess(true);
                    })
                    .catch(() => {
                      setSuccess(false);
                    });
                  setLoading(false);
                  setOpenSnackbar(true);
                }}
                startIcon={<ShoppingCartIcon />}>
                Buy
              </Button>

              <SuccessSnackbar
                successMsg={"Successfully bought!"}
                failMsg={"Transaction failed."}
                success={success}
                open={openSnackbar}
                handleClose={handleClose}
              />

              <Loading isLoading={loading} />
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    </>
  );
}

export default Market;
