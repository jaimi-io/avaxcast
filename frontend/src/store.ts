import { createStore } from "redux";
import { persistStore } from "redux-persist";
import rootReducer from "reducers/";

const store = createStore(rootReducer);
export const persistor = persistStore(store);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export default store;
