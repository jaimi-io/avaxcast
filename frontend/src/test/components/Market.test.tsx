import renderer from "react-test-renderer";
import Market from "components/Market";
import { Provider } from "react-redux";
import store from "store";
import { BrowserRouter as Router } from "react-router-dom";

test("Market snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Market address={"0xdF5C5a3597f2B222a5e62828525624aaAAEB767D"} />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
