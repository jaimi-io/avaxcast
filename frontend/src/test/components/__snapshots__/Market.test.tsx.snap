// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market snapshot 1`] = `
Array [
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <h1
            className="MuiTypography-root MuiTypography-h6"
          >
            Will AVAX/USD be greater than or equal to $X on Wed Sep 08 2021?
          </h1>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                Predicted Price
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                $X
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                End Date
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Wed Sep 08 2021
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                Total Volume
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                0 AVAX
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                Price
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                0 AVAX
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                Your Votes
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                0
              </p>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </span>
              Yes
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                Price
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                0 AVAX
              </p>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                Your Votes
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                0
              </p>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
              No
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            How many shares?
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="number"
              value={0}
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Total Price: 0 AVAX
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              Buy
            </span>
          </button>
          <div
            aria-hidden={true}
            className="MuiBackdrop-root makeStyles-backdrop-2"
            style={
              Object {
                "opacity": 0,
                "visibility": "hidden",
              }
            }
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
